uniffi_bindgen/src/interface/mod.rs

```diff
@@ -1318,44 +1351,100 @@ impl Attributes {
             _ => None,
         })
     }
+}
 
-    fn validate_for_object(&self) -> Result<()> {
-        for attr in self.0.iter() {
-            match attr {
-                Attribute::Threadsafe => continue,
-                _ => bail!(format!("{:?} not supported for Objects", attr)),
-            }
-        }
-        Ok(())
+impl TryFrom<&weedle::attribute::ExtendedAttributeList<'_>> for MethodAttributes {
+    type Error = anyhow::Error;
+    fn try_from(
+        weedle_attributes: &weedle::attribute::ExtendedAttributeList<'_>,
+    ) -> Result<Self, Self::Error> {
+        let attrs = parse_attributes(weedle_attributes, |attr| match attr {
+            Attribute::Throws(_) => Ok(()),
+            _ => bail!(format!("{:?} not supported for methods", attr)),
+        })?;
+        Ok(Self(attrs))
     }
 }
 
-impl Default for Attributes {
-    fn default() -> Self {
-        Attributes(Vec::new())
+// There may be some divergence between Functions and Methods at some point,
+// but not yet.
+type FunctionAttributes = MethodAttributes;
+
+#[derive(Debug, Clone, Hash, Default)]
+pub struct ConstructorAttributes(Vec<Attribute>);
+
+impl ConstructorAttributes {
+    fn get_throws_err(&self) -> Option<&str> {
+        self.0.iter().find_map(|attr| match attr {
+            // This will hopefully return a helpful compilation error
+            // if the error is not defined.
+            Attribute::Throws(inner) => Some(inner.as_ref()),
+            _ => None,
+        })
```

Comment:
It's not obvious to me from reading, why does this need to be its own impl rather than re-using `MethodAttributes`, similar to what we do for `FunctionAttributes` above? In fact I wonder whether we could have *just* the `FunctionAttributes` struct for now and make `Method` and `Constructor` use it directly, until we reach a point where their needs diverge.

Reply:
Oh, good catch. I had done this quite mechanically.

I expect this will only change when we implement #37 .